lib/cdo/redshift_import.rb

```diff
@@ -0,0 +1,105 @@
+require 'cdo/redshift'
+
+class RedshiftImport
+  # Database Migration Service Replication Tasks load data from Aurora into staging Redshift tables with a prefix.
+  TEMP_TABLE_PREFIX = '_import_'.freeze
+  BACKUP_TABLE_PREFIX = '_old_'.freeze
+
+  # DMS Replication Tasks import data into temporary / staging tables prefixed with "_import_".
+  # Complete import of staged data from production database into Redshift by truncating each
+  # target table, moving all of the rows from each import table to its corresponding target table, and then dropping
+  # each temporary / staging import table.
+  # @param schemas [String] Array of Redshift schemas in which to complete data import.
+  def self.complete_table_import(schemas)
+    schemas.each do |schema|
+      temporary_import_tables(schema).each do |import_table|
+        target_table = import_table.partition(TEMP_TABLE_PREFIX).last
+        backup_table = BACKUP_TABLE_PREFIX + target_table
+
+        # Rename existing table to back it up.
+        rename_table(schema, target_table, backup_table)
+
+        # Make staging table the production table.
+        rename_table(schema, import_table, target_table)
+
+        # Rename the primary key from '_import_[foo]_primary' to '[foo]_primary' so that the next time the
```

Comment:
@bencodeorg I've identified an obstacle to the new approach of renaming the staging table.  When the DMS Replication Task imports a table into Redshift, it also creates a primary key on the table.  It seems to have a hard coded name that it chooses for the primary key constraint: `[table_name]_primary`.  Example when importing the MySQL `user_levels` table, it creates the Redshift staging table `_import_user_levels` with a primary key constraint named `_import_user_levels_primary`.

When the staging table is renamed to `user_levels`, the constraint's name is not updated.  The next time we run an import, the DMS Replication Task creates a new `_import_user_levels` table, but then raises a fatal error attempting to create the constraint with the name `_import_user_levels_primary`, because one already exists.

One solution would be to rename the constraint when renaming the staging table, but I can't find a Redshift command for this (Postgres 9.2 seems to support this, but Redshift emulates Postgres 8.x).  As a workaround, we can drop the constraint and create it with the new name, but that might require some additional logic to dynamically determine what the primary key should be (95% of the time it's `id`, but with some tables like `ap_school_codes` it's a composite of multiple columns).

Reply:
After much research and experimentation, I’ve implemented logic to drop and recreate the constraint.